ccm_web/server/src/api/api.interfaces.ts

```diff
@@ -40,7 +40,17 @@ export interface ExternalUserData {
   }
 }
 
-export interface ExternalUserCreationResult {
-  success: boolean
+interface ExternalUserCreationResultBase {
   data: ExternalUserData
 }
+
+interface ExternalUserCreationFailureResult extends ExternalUserCreationResultBase {
+  success: false
+  statusCode: number
+}
+
+interface ExternalUserCreationSuccessResult extends ExternalUserCreationResultBase {
+  success: true
+}
+
+export type ExternalUserCreationResult = ExternalUserCreationFailureResult | ExternalUserCreationSuccessResult
```

Comment:
Interesting that two interfaces can be used to make a third that seems to support conditional structure.  It's much more specific than saying something like…

```ts
success: boolean
statusCode?: number
```

Reply:
Yeah, I've found this really useful in TypeScript. Looks like this would be the documentation explaining it: https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes-func.html#discriminated-unions Honestly not sure when I originally learned about it. See also https://stackoverflow.com/a/59172570